### ISSUE TITLE
[🐛] 🔥 IOS performance traces and other events are not logged correctly in SDK 8.6.0

### ISSUE THREAD

**Issue by dannyhw** [invertase/react-native-firebase#5701]
## Issue

Describe your issue here

In the IOS firebase SDK version 8.6.0 there is a bug that causes some performance traces to not be logged correctly and disappear after some time. This is fixed in 8.6.1 and by overriding the version it resolves the issue. However it would probably be a good idea to update the default version to 8.6.1. From what I could tell the included version is 6.8.0 and I couldn't find a related issue.

I'm happy to make a PR for this if needed. 

see here:
https://firebase.google.com/support/release-notes/ios#version_861_-_august_17_2021

---

### Javascript

<details><summary>Click To Expand</summary>
<p>

#### `package.json`:

```json
  "@react-native-firebase/app": "^12.7.3",
   "@react-native-firebase/perf": "^12.7.3",
```

#### `firebase.json` for react-native-firebase v6:

```json
# N/A
```

</details>

### iOS

<details><summary>Click To Expand</summary>
<p>

#### `ios/Podfile`:

- [ ] I'm not using Pods
- [x] I'm using Pods and my Podfile looks like:

(including $FirebaseSDKVersion = '8.6.1' does fix the issue)

```ruby
require_relative '../node_modules/react-native/scripts/react_native_pods'
require_relative '../node_modules/@react-native-community/cli-platform-ios/native_modules'

platform :ios, '11.0'

target 'testbed' do
  config = use_native_modules!

  use_react_native!(
    :path => config[:reactNativePath],
    # to enable hermes on iOS, change `false` to `true` and then install pods
    :hermes_enabled => true
  )

  target 'testbedTests' do
    inherit! :complete
    # Pods for testing
  end

  # Enables Flipper.
  #
  # Note that if you have use_frameworks! enabled, Flipper will not work and
  # you should disable the next line.
  use_flipper!()

  post_install do |installer|
    react_native_post_install(installer)
  end
end
```

#### `AppDelegate.m`:

```objc
#import "AppDelegate.h"
#import <Firebase.h>
#import <React/RCTBridge.h>
#import <React/RCTBundleURLProvider.h>
#import <React/RCTRootView.h>

#ifdef FB_SONARKIT_ENABLED
#import <FlipperKit/FlipperClient.h>
#import <FlipperKitLayoutPlugin/FlipperKitLayoutPlugin.h>
#import <FlipperKitUserDefaultsPlugin/FKUserDefaultsPlugin.h>
#import <FlipperKitNetworkPlugin/FlipperKitNetworkPlugin.h>
#import <SKIOSNetworkPlugin/SKIOSNetworkAdapter.h>
#import <FlipperKitReactPlugin/FlipperKitReactPlugin.h>

static void InitializeFlipper(UIApplication *application) {
  FlipperClient *client = [FlipperClient sharedClient];
  SKDescriptorMapper *layoutDescriptorMapper = [[SKDescriptorMapper alloc] initWithDefaults];
  [client addPlugin:[[FlipperKitLayoutPlugin alloc] initWithRootNode:application withDescriptorMapper:layoutDescriptorMapper]];
  [client addPlugin:[[FKUserDefaultsPlugin alloc] initWithSuiteName:nil]];
  [client addPlugin:[FlipperKitReactPlugin new]];
  [client addPlugin:[[FlipperKitNetworkPlugin alloc] initWithNetworkAdapter:[SKIOSNetworkAdapter new]]];
  [client start];
}
#endif

@implementation AppDelegate

- (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(NSDictionary *)launchOptions
{
  if ([FIRApp defaultApp] == nil) {
     [FIRApp configure];
   }
#ifdef FB_SONARKIT_ENABLED
  InitializeFlipper(application);
#endif

  RCTBridge *bridge = [[RCTBridge alloc] initWithDelegate:self launchOptions:launchOptions];
  RCTRootView *rootView = [[RCTRootView alloc] initWithBridge:bridge
                                                   moduleName:@"testbed"
                                            initialProperties:nil];

  if (@available(iOS 13.0, *)) {
      rootView.backgroundColor = [UIColor systemBackgroundColor];
  } else {
      rootView.backgroundColor = [UIColor whiteColor];
  }

  self.window = [[UIWindow alloc] initWithFrame:[UIScreen mainScreen].bounds];
  UIViewController *rootViewController = [UIViewController new];
  rootViewController.view = rootView;
  self.window.rootViewController = rootViewController;
  [self.window makeKeyAndVisible];
  return YES;
}

- (NSURL *)sourceURLForBridge:(RCTBridge *)bridge
{
#if DEBUG
  return [[RCTBundleURLProvider sharedSettings] jsBundleURLForBundleRoot:@"index" fallbackResource:nil];
#else
  return [[NSBundle mainBundle] URLForResource:@"main" withExtension:@"jsbundle"];
#endif
}

@end

```

</p>
</details>

---

### Android

<details><summary>Click To Expand</summary>
<p>

#### Have you converted to AndroidX?

- [ ] my application is an AndroidX application?
- [ ] I am using `android/gradle.settings` `jetifier=true` for Android compatibility?
- [ ] I am using the NPM package `jetifier` for react-native compatibility?

#### `android/build.gradle`:

```groovy
// N/A
```

#### `android/app/build.gradle`:

```groovy
// N/A
```

#### `android/settings.gradle`:

```groovy
// N/A
```

#### `MainApplication.java`:

```java
// N/A
```

#### `AndroidManifest.xml`:

```xml
<!-- N/A -->
```

</p>
</details>

---

## Environment

<details><summary>Click To Expand</summary>
<p>

**`react-native info` output:**

```
System:
    OS: macOS 10.15.6
    CPU: (12) x64 Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz
    Memory: 2.33 GB / 16.00 GB
    Shell: 5.7.1 - /bin/zsh
  Binaries:
    Node: 14.17.0 - /usr/local/opt/node@14/bin/node
    Yarn: 1.22.10 - /usr/local/bin/yarn
    npm: 6.14.13 - /usr/local/opt/node@14/bin/npm
    Watchman: 4.9.0 - /usr/local/bin/watchman
  Managers:
    CocoaPods: 1.10.1 - /Users/williamsd2/.rvm/gems/ruby-2.7.0/bin/pod
  SDKs:
    iOS SDK:
      Platforms: iOS 14.4, DriverKit 20.2, macOS 11.1, tvOS 14.3, watchOS 7.2
    Android SDK: Not Found
  IDEs:
    Android Studio: 4.2 AI-202.7660.26.42.7351085
    Xcode: 12.4/12D4e - /usr/bin/xcodebuild
  Languages:
    Java: 15.0.2 - /usr/local/opt/openjdk/bin/javac
  npmPackages:
    @react-native-community/cli: Not Found
    react: 17.0.2 => 17.0.2 
    react-native: 0.64.2 => 0.64.2 
    react-native-macos: Not Found
  npmGlobalPackages:
    *react-native*: Not Found
```

- **Platform that you're experiencing the issue on**:
  - [x] iOS
  - [ ] Android
  - [ ] **iOS** but have not tested behavior on Android
  - [ ] **Android** but have not tested behavior on iOS
  - [ ] Both
- **`react-native-firebase` version you're using that has this issue:**
  - 12.7.3
- **`Firebase` module(s) you're using that has the issue:**
  - `e.g. Instance ID`
- **Are you using `TypeScript`?**
  - Yes

</p>
</details>

---

- 👉 Check out [`React Native Firebase`](https://twitter.com/rnfirebase) and [`Invertase`](https://twitter.com/invertaseio) on Twitter for updates on the library.

**Comment by mikehardy** [comment]
Hi there! You are not wrong about the issue however:

- they did not release 8.6.1 for cocoapods
- if you pod install using the 8.6.0 cocoapods version, and check your Podfile.lock (which you have to recreate on an RNFB update I believe?) I think perf 8.6.1 comes in - that's my experience - do you experience differently?

Which is to say: I don't *think* this is affecting people...that said I'm in no way denying it is an issue, I'm mostly hoping for an 8.7.0 real soon now - just checked now even still not available - and will update as soon as that comes out

**Comment by dannyhw** [comment]
hey @mikehardy I'm able to install the 8.6.1 pods by adding the `$FirebaseSDKVersion = '8.6.1'` line. I was able to produce this issue on a fresh project so I think it's not automatically using 8.6.1, however I could be wrong. 

I'll try removing the pods and podfile.lock and install with 8.6.0 and see what happens.

**Comment by dannyhw** [comment]
Ok I just tried it and it seems like you're right about the performance being installed with 8.6.1. I suppose I just needed to force an update on my podfile.lock

<details><summary>Click To Expand</summary>

```
➜  ios git:(fix/ios-metrics) ✗ rm -rf Podfile.lock Pods               
➜  ios git:(fix/ios-metrics) ✗ pod install
Fetching podspec for `glog` from `../../node_modules/react-native/third-party-podspecs/glog.podspec`
Downloading dependencies
Installing AppsFlyerFramework (6.1.0)
Installing BVLinearGradient (2.5.6)
Installing Braintree (4.37.0)
Installing CocoaAsyncSocket (7.6.5)
Installing DoubleConversion (1.1.6)
Installing FBLazyVector (0.64.2)
Installing FBReactNativeSpec (0.64.2)
Installing FBSDKCoreKit (7.1.1)
Installing FBSDKLoginKit (7.1.1)
Installing FBSDKShareKit (7.1.1)
Installing Firebase (8.6.0)
Installing FirebaseABTesting (8.6.0)
Installing FirebaseAnalytics (8.6.0)
Installing FirebaseCore (8.6.0)
Installing FirebaseCoreDiagnostics (8.6.0)
Installing FirebaseCrashlytics (8.6.0)
Installing FirebaseInstallations (8.6.0)
Installing FirebasePerformance (8.6.1)
Installing FirebaseRemoteConfig (8.6.0)
Installing Flipper (0.96.1)
Installing Flipper-Boost-iOSX (1.76.0.1.11)
Installing Flipper-DoubleConversion (1.1.7)
Installing Flipper-Fmt (7.1.7)
Installing Flipper-Folly (2.6.7)
Installing Flipper-Glog (0.3.6)
Installing Flipper-PeerTalk (0.0.4)
Installing Flipper-RSocket (1.4.3)
Installing FlipperKit (0.96.1)
Installing GCDWebServers (3.5.4)
Installing GamesFramework (7.5.2)
Installing GoogleAnalytics (3.20.0)
Installing GoogleAppMeasurement (8.6.0)
Installing GoogleDataTransport (9.1.0)
Installing GoogleIDFASupport (3.14.0)
Installing GoogleSymbolUtilities (1.1.2)
Installing GoogleTagManager (7.3.1)
Installing GoogleUtilities (7.5.1)
Installing GoogleUtilitiesLegacy (1.3.2)
Installing LiveCasino (21.3.0.14)
Installing OneTrust-CMP-XCFramework (6.21.0.0)
Installing OpenSSL-Universal (1.1.180)
Installing PromisesObjC (2.0.0)
Installing RCT-Folly (2020.01.13.00)
Installing RCTRequired (0.64.2)
Installing RCTTypeSafety (0.64.2)
Installing RNCAsyncStorage (1.12.0)
Installing RNCCheckbox (0.4.2)
Installing RNCMaskedView (0.1.10)
Installing RNCPicker (1.8.0)
Installing RNDateTimePicker (2.4.2)
Installing RNDeviceInfo (5.6.3)
Installing RNFBAnalytics (12.7.5)
Installing RNFBApp (12.7.5)
Installing RNFBCrashlytics (12.7.5)
Installing RNFBPerf (12.7.5)
Installing RNFastImage (8.3.1)
Installing RNGestureHandler (1.10.3)
Installing RNInAppBrowser (3.4.0)
Installing RNLocalize (1.4.3)
Installing RNReanimated (1.9.0)
Installing RNSVG (12.1.0)
Installing RNScreens (2.9.0)
Installing React (0.64.2)
Installing React-Core (0.64.2)
Installing React-CoreModules (0.64.2)
Installing React-RCTActionSheet (0.64.2)
Installing React-RCTAnimation (0.64.2)
Installing React-RCTBlob (0.64.2)
Installing React-RCTImage (0.64.2)
Installing React-RCTLinking (0.64.2)
Installing React-RCTNetwork (0.64.2)
Installing React-RCTSettings (0.64.2)
Installing React-RCTText (0.64.2)
Installing React-RCTVibration (0.64.2)
Installing React-callinvoker (0.64.2)
Installing React-cxxreact (0.64.2)
Installing React-jsi (0.64.2)
Installing React-jsiexecutor (0.64.2)
Installing React-jsinspector (0.64.2)
Installing React-perflogger (0.64.2)
Installing React-runtimeexecutor (0.64.2)
Installing ReactCommon (0.64.2)
Installing SDWebImage (5.11.1)
Installing SDWebImageWebPCoder (0.6.1)
Installing Stripe (21.8.1)
Installing StripeCore (21.8.1)
Installing TouchID (4.4.1)
Installing Yoga (1.14.0)
Installing YogaKit (1.18.1)
Installing boost-for-react-native (1.63.0)
Installing glog (0.3.5)
Installing hermes-engine (0.7.2)
Installing ios-settings-bundle (1.2.2)
Installing libevent (2.1.12)
Installing libwebp (1.2.0)
Installing nanopb (2.30908.0)
Installing react-native-appsflyer (6.1.0)
Installing react-native-cookies (5.0.1)
Installing react-native-fbsdk (2.0.0)
Installing react-native-flipper (0.96.1)
Installing react-native-safe-area-context (3.1.1)
Installing react-native-sensitive-info (5.5.8)
Installing react-native-startup-time (2.0.0)
Installing react-native-startup-trace (0.4.0)
Installing react-native-user-agent (2.1.0)
Installing react-native-webview (11.3.2)
Generating Pods project
Patching xxx/native/ios/Pods/RCT-Folly/folly/synchronization/DistributedMutex-inl.h
Integrating client project
```
</details>

**Comment by mikehardy** [comment]
Okay - that matched my results and gives me a little more comfort but I'm still not 100% happy either. I think the window of when 8.6.0 came out vs 8.6.1 was pretty small at least.

I see why you were able to reference 8.6.1 as a cocoapod dependency - I misremembered above and was wrong there - they issued cocoapods but they did not issue the *zip* builds, and for us that is a little problematic because we have a lot of people that have integrated https://github.com/invertase/firestore-ios-sdk-frameworks via the recommended method of a version reference to the package version, but those frameworks depend on the zip build etc so if we issued an 8.6.1 here (using cocoapods) but there was no zip build (for the frameworks pre-build) we would break their build as collateral damage.

I'm not sure what the best course is (other than an 8.7.0 with full release!). There is a longer term effort internal to the official firebase-ios-sdk format to distribute pre-builds like our firestore-ios-sdk-frameworks so this window of difficult-patch-releases will be closed in a future state I hope
